Use assertj fluent style in flowable-bpmn-converter module.

dbmalkovsky opened the pull request:
This is part 5 of 5 parts.

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/SubProcessConverterAutoLayoutTest.java:
```diff
@@ -12,11 +12,8 @@
-
-import java.util.List;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.tuple;
```
Please use only `Assertions`.

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/SubProcessConverterAutoLayoutTest.java:
```diff
@@ -54,32 +51,31 @@ protected String getResource() {
+                    assertThat(subProcess.getDataObjects())
+                            .extracting(ValuedDataObject::getName, ValuedDataObject::getValue)
+                            .containsExactly(tuple("SubTest", "Testing"));
+                    assertThat(subProcess.getDataObjects().get(0).getItemSubjectRef().getStructureRef())
```
No need for `isInstanceOfSatisfying` you can directly do `isEqualTo("xsd:string")`. If it is not a string then it won't be equal to a string as well

dbmalkovsky replied:
Done.

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/SubProcessWithExtensionsConverterTest.java:
```diff
@@ -131,43 +128,37 @@ protected String getResource() {
-                fail("Unknown key value");
-            }
-        }
+        assertThat(attributes).containsOnlyKeys("Attr3", "Attr4");
+        assertThat(attributes).containsEntry("Attr3", "3");
+        assertThat(attributes).containsEntry("Attr4", "4");
```
This can also look like:

```
assertThat(attribute).containsOnly(
    entry("Attr3", "3"),
    entry("Attr4", "4")
)
```

dbmalkovsky replied:
Done

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/UserTaskConverterTest.java:
```diff
@@ -49,83 +45,58 @@ protected String getResource() {
-        assertTrue(userTask.getCustomGroupIdentityLinks().get("manager").contains("management"));
-        assertTrue(userTask.getCustomGroupIdentityLinks().get("businessAdministrator").contains("management"));
+                    assertThat(userTask.getAssignee()).isEqualTo("kermit");
+                    assertThat(userTask.getCandidateUsers()).hasSize(2);
+                    assertThat(userTask.getCandidateUsers().contains("kermit")).isTrue();
+                    assertThat(userTask.getCandidateUsers().contains("fozzie")).isTrue();
```
```suggestion
                    assertThat(userTask.getCandidateUsers()).containsExactlyInAnyOrder("kermit", "fozzie");
```

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/UserTaskConverterTest.java:
```diff
@@ -49,83 +45,58 @@ protected String getResource() {
+                    assertThat(userTask.getCandidateUsers()).hasSize(2);
+                    assertThat(userTask.getCandidateUsers().contains("kermit")).isTrue();
+                    assertThat(userTask.getCandidateUsers().contains("fozzie")).isTrue();
+                    assertThat(userTask.getCandidateGroups()).hasSize(2);
+                    assertThat(userTask.getCandidateGroups().contains("management")).isTrue();
+                    assertThat(userTask.getCandidateGroups().contains("sales")).isTrue();
```
```suggestion
                    assertThat(userTask.getCandidateGroups()).containsExactlyInAnyOrder("management", "sales);
```

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/UserTaskConverterTest.java:
```diff
@@ -49,83 +45,58 @@ protected String getResource() {
-        assertTrue(StringUtils.isEmpty(formProperty.getVariable()));
-        assertTrue(StringUtils.isEmpty(formProperty.getExpression()));
-        assertEquals(2, formProperty.getFormValues().size());
+                    assertThat(userTask.getCustomUserIdentityLinks()).hasSize(1);
+                    assertThat(userTask.getCustomGroupIdentityLinks()).hasSize(2);
+                    assertThat(userTask.getCustomUserIdentityLinks().get("businessAdministrator").contains("kermit")).isTrue();
+                    assertThat(userTask.getCustomGroupIdentityLinks().get("manager").contains("management")).isTrue();
+                    assertThat(userTask.getCustomGroupIdentityLinks().get("businessAdministrator").contains("management")).isTrue();
 
```
```suggestion
                    assertThat(userTask.getCustomUserIdentityLinks()).containsOnly(entry("businessAdministrator", Collections.singleton("kermit")));
                    assertThat(userTask.getCustomGroupIdentityLinks()).containsOnly(
                        entry("manager", Collections.singleton("management")),
                        entry("businessAdministrator", Collections.singleton("management"))
                    );
```

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/ValuedDataObjectConverterTest.java:
```diff
@@ -61,148 +58,148 @@ protected String getResource() {
+        assertThat(dataObj.getName()).isEqualTo("LongTest");
+        assertThat(dataObj.getItemSubjectRef().getStructureRef()).isEqualTo("xsd:long");
+        assertThat(dataObj.getValue()).isInstanceOf(Long.class);
+        assertThat(dataObj.getValue()).isEqualTo((long) -123456);
```
No need for cast you can write it like `-123456L` instead

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/ValuedDataObjectWithExtensionsConverterTest.java:
```diff
@@ -131,96 +128,82 @@ protected String getResource() {
-                fail("Unknown key value");
-            }
-        }
+        assertThat(attributes).containsOnlyKeys("Attr1", "Attr2");
+        assertThat(attributes).containsEntry("Attr1", "1");
+        assertThat(attributes).containsEntry("Attr2", "2");
```
```suggestion
        assertThat(attributes).containsOnly(
            entry("Attr1", "1"),
            entry("Attr2", "2")
        );
```

filiphr commented on modules/flowable-bpmn-converter/src/test/java/org/flowable/editor/language/xml/ValuedDataObjectWithExtensionsConverterTest.java:
```diff
@@ -131,96 +128,82 @@ protected String getResource() {
-                fail("Unknown key value");
-            }
-        }
+        assertThat(attributes).containsOnlyKeys("Attr3", "Attr4");
+        assertThat(attributes).containsEntry("Attr3", "3");
+        assertThat(attributes).containsEntry("Attr4", "4");
```
```suggestion
        assertThat(attributes).containsOnly(
            entry("Attr3", "3"),
            entry("Attr4", "4"),
        );
```

dbmalkovsky commented:
@filiphr I think I've addressed all your comments.  Thank you much for the review.

BTW, sometimes I didn't change things (like the check of the string class) because that was how it was originally coded.

filiphr commented:
Thanks @dbmalkovsky.

>BTW, sometimes I didn't change things (like the check of the string class) because that was how it was originally coded.

Yes I noticed that. I think that it is not a problem to change it sometimes if the test stays the same.
